forge-game/src/main/java/forge/game/ForgeScript.java

```diff
@@ -91,6 +92,29 @@ public static boolean cardStateHasProperty(CardState cardState, String property,
                 default:
                     return false;
             }
+        } else if (property.startsWith("ChosenLetter")) {
+            final List<String> letters = source.getChosenLetters();
+            if (letters == null) return false; // some things check before letters have been chosen
+            String name = cardState.getName();
```

Comment:
Maybe apply `.toUpperCase` to `name` here so you don't have to check it twice below? Also in case we ever have any cards that start with a lowercase letter. Can't think of any right now besides "capital offense".